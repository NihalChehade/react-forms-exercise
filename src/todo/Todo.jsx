import React from "react";

const Todo = ({id, value, removeTodo }) => {
  return <li>{value}<button onClick={()=>{removeTodo(id)}}>X</button></li>;
};

export default Todo;
