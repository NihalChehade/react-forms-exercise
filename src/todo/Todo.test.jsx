import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';

// Smoke Test
test('renders Todo component without crashing', () => {
  render(<Todo id="1" value="Test Task" removeTodo={() => {}} />);
});

// Snapshot Test
test('matches snapshot', () => {
  const { asFragment } = render(<Todo id="1" value="Test Task" removeTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic Test
test('calls removeTodo function with correct id when button is clicked', () => {
  const mockRemoveTodo = vi.fn();
  render(<Todo id="1" value="Test Task" removeTodo={mockRemoveTodo} />);
  
  const button = screen.getByText('X');
  fireEvent.click(button);
  
  expect(mockRemoveTodo).toHaveBeenCalledWith('1');
});
