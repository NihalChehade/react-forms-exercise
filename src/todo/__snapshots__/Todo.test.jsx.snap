// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    Test Task
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
